### DISCUSSION TITLE
Add Custom Attribute To gutenberg block

### DISCUSSION THREAD

**Discussion by dthenley** [WordPress/gutenberg#37117] [Q&A]
I primarily dev themes for my clients and bake most of the styling they would need into the site. I am trying to stay away from making a custom ACF block for everything when i could just as well add a certain class to achieve the same result. I am having some issue with trying to add custom attribute to core blocks. When theming what do you think is the best approach to take to add a custom attribute on a gutenberg block?

**Comment by Lovor01** [comment]
There is possibility to add custom classes in the block editor in advanced part of the inspector, and also, you can target specific block by its name, namely `.wp-block-block-name`, but if you really need to add class programmatically, you can look into [Block filters](https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters). You would want to use `blocks.getSaveContent.extraProps` filter.
